datavault-common/src/main/java/org/datavaultplatform/common/model/dao/RoleAssignmentDaoImpl.java

```diff
@@ -16,64 +13,93 @@
 
     private SessionFactory sessionFactory;
 
-    public RoleAssignmentDaoImpl() {}
+    public RoleAssignmentDaoImpl() {
+    }
 
     public void setSessionFactory(SessionFactory sessionFactory) {
         this.sessionFactory = sessionFactory;
     }
 
     @Override
     public boolean roleAssignmentExists(RoleAssignment roleAssignment) {
-        Session session = this.sessionFactory.openSession();
-        Criteria criteria = session.createCriteria(RoleAssignment.class);
-        criteria.add(Restrictions.eq("role", roleAssignment.getRole()));
-        criteria.add(Restrictions.eq("user", roleAssignment.getUser()));
-        if (roleAssignment.getSchool() != null) {
-            criteria.add(Restrictions.eq("school", roleAssignment.getSchool()));
-        }
-        if (roleAssignment.getVault() != null) {
-            criteria.add(Restrictions.eq("vault", roleAssignment.getVault()));
+        Session session = null;
+        try {
+            session = this.sessionFactory.openSession();
+
+            Criteria criteria = session.createCriteria(RoleAssignment.class);
+            criteria.add(Restrictions.eq("role", roleAssignment.getRole()));
+            criteria.add(Restrictions.eq("user", roleAssignment.getUser()));
+            if (roleAssignment.getSchool() != null) {
+                criteria.add(Restrictions.eq("school", roleAssignment.getSchool()));
```

Comment:
This data model strikes me as very odd. There's nothing stopping a row existing with both a `school` and a `vault` and if I have to null check before using the result of either call then the API isn't very nice.

String literals should be defined as constants.

Out of scope for this task, though, just commentary on what's been added.